choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerDBService.java

```diff
@@ -264,6 +376,16 @@ public ChoreographerAction addNextActionToAction(final String name, final String
         }
     }
 
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerPlanResponseDTO createPlanResponse(final String name, final String firstActionName, final List<ChoreographerActionRequestDTO> actions) {
+        logger.debug("createPlanResponse started...");
+
+        final ChoreographerPlan planEntry = createPlan(name, firstActionName, actions);
+
+        return DTOConverter.convertPlanToPlanResponseDTO(planEntry);
+    }
+
 	//-------------------------------------------------------------------------------------------------
 	@Transactional(rollbackFor = ArrowheadException.class)
     public ChoreographerPlan createPlan(final String name, final String firstActionName, final List<ChoreographerActionRequestDTO> actions) {
```

Comment:
need to create a plan verifier that checks:

- firstActionName empty check
- action list not null or empty check
- action list contains an action with name specified in firstActionName
- and formal checks on actions and steps and service definitions, version requirements
- name duplications in actions and steps
- circle references in actions and steps
- references to parallel steps